### PR TITLE
Terraform 0.14 upgrade

### PR DESCRIPTION
**Pull request by maximmi**
## what
- Apply Cloud Posse standard updates for Terraform 0.14

## why
- Make module library compatible with Terraform 0.14

### CODE REVIEW

File: .github/workflows/auto-update.yml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Update context.tf
+        shell: bash
+        id: update
+        env:
+          GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+        run: |
+          if [[ -f context.tf ]]; then
+            echo "context.tf exist! Fetching most recent version to see if there is an update"
```
Comment (osterman) [inline .github/workflows/auto-update.yml:21]:
```suggestion
            echo "Discovered existing context.tf! Fetching most recent version to see if there is an update."
```

File: .github/workflows/auto-update.yml [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            curl -o context.tf -s https://raw.githubusercontent.com/cloudposse/terraform-null-label/master/exports/context.tf
```
Comment (osterman) [inline .github/workflows/auto-update.yml:22]:
```suggestion
            curl -o context.tf -fsSL https://raw.githubusercontent.com/cloudposse/terraform-null-label/master/exports/context.tf
```

File: .github/workflows/auto-update.yml [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            if git diff --no-patch --exit-code context.tf; then
+              echo "No changes detected! Exiting the job..."
+              echo "MAKE_PR=false" >> $GITHUB_ENV
+              exit 0
+            else
+              echo "context.tf file was updated. Proceeding with PR..."
+              echo "MAKE_PR=true" >> $GITHUB_ENV
+              exit 0
+            fi
```
Comment (osterman) [inline .github/workflows/auto-update.yml:31]:
```suggestion
            else
              make init
              make readme
              exit 0
            fi
```

File: .github/workflows/auto-update.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+          else
+            echo "context.tf does not exist! It is not enough to fetch context.tf, module requires to be updated to support it."
+            echo "MAKE_PR=false" >> $GITHUB_ENV
+            exit 0
+          fi
+
+      - name: Create Pull Request
+        if: ${{ success() && env.MAKE_PR == 'true' }}
```
Comment (osterman) [inline .github/workflows/auto-update.yml:39]:
```suggestion
```

File: .github/workflows/auto-update.yml [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        uses: cloudposse/actions/github/create-pull-request@0.20.0
+        with:
+          token: ${{ secrets.PUBLIC_REPO_ACCESS_TOKEN }}
+          commit-message: Update context.tf from origin source
+          title: Update context.tf
+          body: |-
+            ## what
+            This is an auto-generated PR that updates the context.tf file
```
Comment (osterman) [inline .github/workflows/auto-update.yml:47]:
```suggestion
            This is an auto-generated PR that updates the context.tf file to the latest version from `cloudposse/terraform-null-label`
```

File: .github/workflows/auto-update.yml [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+            ## why
+            To have most recent changes of `context`
```
Comment (osterman) [inline .github/workflows/auto-update.yml:50]:
```suggestion
            To support all the features of the `context` interface.
```

File: .github/workflows/auto-update.yml [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+              echo "MAKE_PR=false" >> $GITHUB_ENV
+              exit 0
+            else
+              echo "context.tf file was updated. Proceeding with PR..."
+              echo "MAKE_PR=true" >> $GITHUB_ENV
+              exit 0
+            fi
+          else
+            echo "context.tf does not exist! It is not enough to fetch context.tf, module requires to be updated to support it."
+            echo "MAKE_PR=false" >> $GITHUB_ENV
```
Comment (osterman) [inline .github/workflows/auto-update.yml:34]:
```suggestion
```

File: .github/workflows/auto-update.yml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+              echo "No changes detected! Exiting the job..."
+              echo "MAKE_PR=false" >> $GITHUB_ENV
+              exit 0
+            else
+              echo "context.tf file was updated. Proceeding with PR..."
+              echo "MAKE_PR=true" >> $GITHUB_ENV
+              exit 0
+            fi
+          else
+            echo "context.tf does not exist! It is not enough to fetch context.tf, module requires to be updated to support it."
```
Comment (osterman) [inline .github/workflows/auto-update.yml:33]:
```suggestion
            echo "This module has not yet been updated to support the context.tf pattern! Please update in order to support automatic updates."
```

File: .github/workflows/auto-update.yml [hunk 9]
```diff
@@ -0,0 +1,92 @@
+name: "auto-update"
+on:
+  schedule:
+    # Update context.tf and README.md nightly
+    - cron:  '0 0 * * *'
+
+jobs:
+  context-update:
```
Comment (osterman) [inline .github/workflows/auto-update.yml:8]:
```suggestion
  context.tf:
```

File: .github/workflows/auto-context.yml [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+
+    - name: Update context.tf
+      shell: bash
+      id: update
+      env:
+        GITHUB_TOKEN: "${{ secrets.GITHUB_TOKEN }}"
+      run: |
+        if [[ -f context.tf ]]; then
+          echo "Discovered existing context.tf! Fetching most recent version to see if there is an update."
+          curl -o context.tf -fsSL https://raw.githubusercontent.com/cloudposse/terraform-null-label/master/exports/context.tf
```
Comment (osterman) [inline .github/workflows/auto-context.yml:22]: Should this call the context target from the build-harness?

Comment (Nuru) [reply .github/workflows/auto-context.yml:22]: Given that this is run 200 times nightly, I want to keep it light and not require build-harness. I will call build-harness if `context.tf` changes.

**Review by Nuru — APPROVED** [review]

**Pull request merged**